dali/kernels/reduce/mean_stddev_gpu_impl.cuh

```diff
@@ -0,0 +1,386 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_REDUCE_MEAN_STDDEV_GPU_IMPL_CUH_
+#define DALI_KERNELS_REDUCE_MEAN_STDDEV_GPU_IMPL_CUH_
+
+#include "dali/kernels/reduce/reduce_gpu_impl.cuh"
+#include "dali/kernels/reduce/reduce_drop_dims.h"
+
+namespace dali {
+namespace kernels {
+namespace reduce_impl {
+
+template <typename Out, typename Scale>
+struct ScaleAndConvert {
```

Comment:
I feel like this belongs into the MeanImplBase, it's used only there, and those two are completely tied together with how they are implemented.

Reply:
This sounds pretty generic. It can be moved to some utils as well.